contracts/AssetPool.sol

```diff
@@ -1,41 +1,56 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.4.22 <0.9.0;
+pragma solidity ^0.8.0;
 import "./LiquidityPoolInterfaces.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "hardhat/console.sol";
+import "./Proxy.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 
-contract AssetPool is LiquidityPoolInterfaces, Ownable, ReentrancyGuard, ERC20("Cruize Protection LP Token", "cr") {
+contract AssetPool is Proxy, Ownable, ReentrancyGuard {
   using SafeMath for uint256;
-  mapping(address => mapping(address => uint)) public userInfo;
+  mapping(address => mapping(address => uint256)) public userInfo;
   mapping(address => bool) private allowedAssets;
+  mapping(address => address) public assetToLp;
+  address[] public tokenAddress;
+
+
+  event Withdraw(
+      address indexed account,
+      uint256 writeAmount
+  );
+  event Provide(address indexed account,uint256 writeAmount);
+
+  // This is proxy code
+  address immutable masterContract;
+
+  constructor(address _masterContract) {
+    masterContract = _masterContract;
+  }
+
+  function createToken(string memory name_, string memory symbol_, address assetAddress) external onlyOwner nonReentrant {
+    CruizeToken token = CruizeToken(createClone(masterContract));
+    token.init(name_, symbol_);
+    assetToLp[assetAddress] = address(token);
+  }
+  // Proxy code ends here
 
   function depositAsset(uint256 depositAmount, address assetAddress) external payable nonReentrant {
     require(depositAmount > 0, "1: Deposit Amount cannot be 0");
-    require(allowedAssets[assetAddress] == true, "Asset not approved for deposit");
-    userInfo[msg.sender][assetAddress] += (depositAmount);
-    IERC20 depositToken = IERC20(assetAddress);
+    userInfo[msg.sender][assetAddress] += depositAmount;
+    ILPtoken depositToken = ILPtoken(assetAddress);
```

Comment:
You have to put a check condition here for the deposit asset that is we are allowed to deposit this asset or not?
for ex. someone says i want to put the link but we. are not allowing anyone to deposit link's that condition must be checked here. before accessing the user's assets

Reply:
yeah basically line 12 was supposed to do that. We should not have removed it I guess ?

Reply:
we have managed that thing.Karna does not put the updated code